src/galileo/datasets.py

```diff
@@ -74,6 +74,22 @@ def get_content(self) -> Union[None, DatasetContent]:
 
         return content
 
+    def _get_etag(self) -> Optional[str]:
```

Comment:
can you write unit test for this new function?

Reply:
Done.